srtcore/core.cpp

```diff
@@ -8706,7 +9175,7 @@ void CUDT::checkACKTimer(uint64_t currtime_tk)
 
 void CUDT::checkNAKTimer(uint64_t currtime_tk)
 {
-    if (!m_bRcvNakReport)
+    if (!m_bRcvNakReport || m_PktFilterRexmitLevel != SRT_ARQ_ALWAYS)
```

Comment:
Ok, I think I remember the problem. It was because there was one unsolved problem with NAKREPORT such that the loss report is practically always sent immediately twice (NAKREPORT starts with sending it, while there's no timer registerred at the loss, so the NAKREPORT doesn't know when last time the report was sent).

The real reason as to why sending NAKREPORT with losses reported by the packet filter (it's only sent once) was because of the structure of the loss list. It was predicted to contain losses sorted in the sequence order and it was using a special case that losses recorded as they happen will always keep the order of insertion in sync with the order of sequences. Losses reported by the packet filter, however, don't keep this rule and therefore recording these losses with the current structure of the loss list would require lots of work to adjust this container to these needs, and with this kind of structure "one does not simply" add these losses to the list.

So, it's not something carved in the stone that in `ARQ_ONREQ` mode NAKREPORT can't work, only this requires some effort to be put into it.

I'll put appropriate explanations in the comment.